src/chromo/models/pythia8.py

```diff
@@ -161,28 +181,60 @@ def _cross_section(self, kin=None):
             diffractive_axb=st.sigmaAXB,
         )
 
+        # TODO use pythia.info.sigmaGen(i) for cross-sections including nucleus in the
+        # initial collision system. pythia.info.sigmaTot() are the estimated
+        # cross-section values for the equivalent pp collision for each sub-processes.
+        # pythia.info.sigmaGen(i) are the cross-section values generated by the MC
+        # events for each subprocesses. Best is to grab these values at the end of the
+        # event generation when the full statistics is available. The subprocesses are
+        # defined with codes in Pythia:
+        #   i    subprocess
+        #   101  non-diffractive
+        #   102  A B -> A B elastic
+        #   103  A B -> X B single diffractive
+        #   104  A B -> A X single diffractive
+        #   105  A B -> X X double diffractive
+        #   106  A B -> A X B central diffractive
+
     def _set_kinematics(self, kin):
         config = self._config[:]
 
+        # Path prefix for some cache files that Pythia8 generates
+        # to speed up subsequent runs
+        cache_prefix = str(
+            _cache_base_dir()
+            / "Pythia8"
+            / f"cache_{int(kin.p1)}_{int(kin.p2)}_{kin.ecm / MeV:.0f}".replace("-", "m")
+        )
+
         # TODO use numpy PRNG instead of Pythia's
         config += [
             # use center-of-mass frame
             "Beams:frameType = 1",
             # do not print progress
             "Next:numberCount = 0",
+            "MultipartonInteractions:reuseInit = 0",
```

Comment:
Doesn't this disable the cache permanently for this instance of the Pythia8 class? There is nothing to reset it back to the value 3

Reply:
It was a mistake on my side, sorry!